lib/puppet/provider/service/smf.rb

```diff
@@ -135,8 +143,8 @@ def status
       states = service_states
       state = states[1] == "-" ? states[0] : states[1]
     rescue Puppet::ExecutionFailure
-      info "Could not get status on service #{self.name}"
-      return :stopped
+      debug "Could not get status on service #{self.name} #{$!}"
+      return :absent
```

Comment:
@MosesMendoza dropped exists. If that's good I'll squash the commits unless you now use the new-ish "squash commits" option on merge.